api/overdrive.py

```diff
@@ -78,13 +79,15 @@ def patron_request(self, patron, pin, url, extra_headers={}, data=None,
         headers = dict(Authorization="Bearer %s" % patron_credential.credential)
         headers.update(extra_headers)
         if method and method.lower() in ('get', 'post', 'put', 'delete'):
-            method = getattr(requests, method.lower())
+            method = method.lower()
         else:
             if data:
-                method = requests.post
+                method = 'post'
             else:
-                method = requests.get
-        response = method(url, headers=headers, data=data)
+                method = 'get'
+        response = HTTP.request_with_timeout(
```

Comment:
This is OverdriveAPI.patron_request. This code ensures that if we make any request using a patron's access token, and we get an error, the patron sees a 502 error rather than a 500.